packages/relay/src/lib/eth.ts

```diff
@@ -473,8 +481,16 @@ export class EthImpl implements Eth {
     if (blockNumber === 0) {
       return '0x0';
     } else {
-      const accountInfo = await this.sdkClient.getAccountInfo(address);
-      return EthImpl.numberTo0x(Number(accountInfo.ethereumNonce));
+      let nonce = 0;
+      const result = await this.mirrorNodeClient.resolveEntityType(address);
+      if (result?.type === 'account') {
+        const accountInfo = await this.sdkClient.getAccountInfo(result?.entity.account);
+        nonce = Number(accountInfo.ethereumNonce);
+      } else if (result?.type === 'contract') {
+        nonce = 1;
```

Comment:
We will have a problem with Contract `nonces`. 
Nonces for a contract is `1 + n` where `n` is the number of times the contract _created other contracts through CREATE or CREATE2_. The deployment of the contract itself is considered part of the `nonce`, that's why the nonce starts from `1`.

Since we are not able to query `getAccountInfo` for contracts, I think the best option is to hardcode the value to `1` and add TODO comment that "currently contract nonces are hardcoded to 1, thus factory contracts will not get their `transactionCount` updated/reflected" 

The long-term solution IMO is to update the nonce of the `contract` for every `CREATE/CREATE2` opcode executed from it and expose it to mirror nodes as information.

@shemnon do you share the same opinion?

Reply:
Interesting. I'm okay with the constant for now.
Does that in itself cause errors or blockers?

Also instead of a TODO comment please open a ticket capturing the details of the issue with your suggested solution or other options.